### PR TITLE
[KYUUBI#4530] fix data masking  does not support CJK for strategies such as MASK_SHOW_FIRST_4 or MASK_SHOW_FIRST_4

### PR DESCRIPTION
**Pull request by Jackhjf**
…es such as MASK_SHOW_FIRST_4 or MASK_SHOW_FIRST_4

### _Why are the changes needed?_

The data masking  does not support Chinese for strategies such as MASK_SHOW_FIRST_4 or MASK_SHOW_FIRST_4
image

### _How was this patch tested?_
create table iceberg_spark.test_copy(id string,name string);
insert into  iceberg_spark.test_copy values(“111111”，“中国人人人儿呢热");
insert into  iceberg_spark.test_copy values(“222223”，“rrrrxxxxxx");

Configure the ranger strategy
image
image

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala [hunk 1]
```diff
@@ -135,7 +135,8 @@ object SparkRangerAdminPlugin extends RangerBasePlugin("spark", "sparkSql")
     val pos = if (hasLen) ", 5" else ""
     val upper = s"regexp_replace($expr, '[A-Z]', 'X'$pos)"
     val lower = s"regexp_replace($upper, '[a-z]', 'x'$pos)"
-    val digits = s"regexp_replace($lower, '[0-9]', 'n'$pos)"
+    val chinese = s"regexp_replace($lower, '[\u4e00-\u9fa5]', 'x'$pos)"
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:138]: What's the behavior on other popular databases/OLAP systems?
And I remember that UTF-8 reserved range for CJK, can we extend it to cover CJK?

Comment (Jackhjf) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala]: > What's the behavior on other popular databases/OLAP systems? And I remember that UTF-8 reserved range for CJK, can we extend it to cover CJK?
Spark needs to access the metastore to query data, and the metadata information in the metastore is stored in mysql.In mysql, the character set is also utf-8
image

CJK can be covered by the following regular expression
image

**Comment by yaooqinn** [comment]
Do we need a custom masker here instead? the hive masker udfs only support alphabets&digits

**Comment by yaooqinn** [comment]
closed via #4643

**Pull request closed without merging**